### PR TITLE
APM-330003-set-one-sh-lib

### PR DESCRIPTION
**Pull request by karolkieglerski**

### CODE REVIEW

File: scripts/lib.sh [hunk 1]
```diff
@@ -34,13 +35,21 @@ err() {
...
-clean_extensions() {
+clean() {
   echo "- removing extensions files"
   rm -rf $EXTENSION_MANIFEST_FILE $CLUSTER_EXTENSIONS_TMPDIR $EXTENSIONS_TMPDIR
+
+  if [ -n "$GCP_FUNCTION_NAME" ]; then
+    echo "- removing archive [$FUNCTION_ZIP_PACKAGE]"
+    rm $WORKING_DIR/$FUNCTION_ZIP_PACKAGE
+
+    echo "- removing temporary directory [$FUNCTION_ZIP_PACKAGE]"
```
Comment (mswiatkowska) [inline scripts/lib.sh:46]: (minor) please change $FUNCTION_ZIP_PACKAGE to $GCP_FUNCTION_NAME

**Review by mswiatkowska — APPROVED** [review]

File: scripts/lib.sh [hunk 2]
```diff
@@ -158,6 +176,36 @@ check_api_token() {
...
 
+check_s3_url() {
+  if [ -z "$EXTENSION_S3_URL" ]; then
+    EXTENSION_S3_URL="https://dynatrace-gcp-extensions.s3.amazonaws.com"
+  else
+    warn "Development mode on: custom S3 url link."
+  fi
+}
+
+check_gcp_config_in_extension() {
```
Comment (Dawid-Kaszubski) [inline scripts/lib.sh:187]: After this PR: https://github.com/dynatrace-oss/dynatrace-gcp-function/pull/209
`check_gcp_config_in_extension` and `check_gcp_config_in_extensions` should be replaced by `validate_gcp_config_in_extensions` function

File: scripts/lib.sh [hunk 3]
```diff
@@ -245,18 +287,21 @@ get_extensions_from_dynatrace() {
...
     EXTENSION_NAME="$(cut -d':' -f1 <<<"${EXTENSIONS_FROM_CLUSTER_ARRAY[$i]}")"
     EXTENSION_VERSION="$(cut -d':' -f2 <<<"${EXTENSIONS_FROM_CLUSTER_ARRAY[$i]}")"
-    
+
     curl -k -s -X GET "${DYNATRACE_URL}/api/v2/extensions/${EXTENSION_NAME}/${EXTENSION_VERSION}" -H "Accept: application/octet-stream" -H "Authorization: Api-Token ${DYNATRACE_ACCESS_KEY}" -o "${EXTENSION_NAME}-${EXTENSION_VERSION}.zip"
     if [ -f "${EXTENSION_NAME}-${EXTENSION_VERSION}.zip" ] && [[ "$EXTENSION_NAME" =~ ^com.dynatrace.extension.(google.*)$ ]]; then
-      find ${EXTENSIONS_TMPDIR} -regex ".*${BASH_REMATCH[1]}.*" -exec rm -rf {} \;
-      mv "${EXTENSION_NAME}-${EXTENSION_VERSION}.zip" ${EXTENSIONS_TMPDIR}
+      check_gcp_config_in_extension "${EXTENSION_NAME}-${EXTENSION_VERSION}.zip"
+      if [ -f "${EXTENSION_NAME}-${EXTENSION_VERSION}.zip" ]; then
```
Comment (Dawid-Kaszubski) [inline scripts/lib.sh:298]: `check_gcp_config_in_extension` call and if statement should be removed.
Lines 299 and 300 should stay.

Comment (karolkieglerski) [reply scripts/lib.sh:301]: the line numeration is a little bit different, this is ok? https://github.com/dynatrace-oss/dynatrace-gcp-function/pull/214/commits/ee73793f2a1960f461bd15364a9ecba9fa23d7c4

**Review by Dawid-Kaszubski — APPROVED** [review]

**Review by sq2gxo — APPROVED** [review]

**Review by mswiatkowska — APPROVED** [review]

**Review by Dawid-Kaszubski — APPROVED** [review]

**Pull request merged**